lib/bb/build.py: 'deps' may not always exist in a few corner cases.

With a small set of recipes (1), and all system dependencies assume_provided,
we may end up with no parent dependencies.  So we check if 'deps' was defined
instead of always assuming it exists.

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

diff -ur bitbake/lib/bb/build.py bitbake/lib/bb/build.py
--- bitbake/lib/bb/build.py	2013-01-03 16:14:12.000000000 -0600
+++ bitbake/lib/bb/build.py	2013-01-05 16:34:05.155737187 -0600
@@ -588,9 +588,10 @@
         getTask('noexec')
         getTask('umask')
         task_deps['parents'][task] = []
-        for dep in flags['deps']:
-            dep = data.expand(dep, d)
-            task_deps['parents'][task].append(dep)
+        if 'deps' in flags:
+            for dep in flags['deps']:
+                dep = data.expand(dep, d)
+                task_deps['parents'][task].append(dep)
 
     # don't assume holding a reference
     data.setVar('_task_deps', task_deps, d)
